$ OpenBSD: $

1) We prefer to use the system malloc instead of jemalloc. The problem
that is solved with it has never been reproduced with OpenBSD malloc.
more: https://github.com/telegramdesktop/tdesktop/issues/16084

2) Avoid unnecessary dependence on libatomic.
According to clang docs, compiler-rt contains an implementation of
an atomics library, so libatomic from gcc is rendundant.
more: https://clang.llvm.org/docs/Toolchain.html#atomics-library

The ability to build with ported gcc, probably requires a
different version of the patch.

3) Turn on '-pie' - the bug that caused it to be disabled is solved.
more: https://gitlab.gnome.org/GNOME/nautilus/-/issues/1601

Index: cmake/options_linux.cmake
--- cmake/options_linux.cmake.orig
+++ cmake/options_linux.cmake
@@ -7,7 +7,6 @@
 target_compile_options(common_options
 INTERFACE
     -fstack-protector-all
-    -fstack-clash-protection
     -fPIC
     $<IF:$<CONFIG:Debug>,,-fno-strict-aliasing>
     -pipe
@@ -61,11 +60,6 @@ if (DESKTOP_APP_SPECIAL_TARGET)
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
 endif()
 
-target_link_libraries(common_options
-INTERFACE
-    desktop-app::external_jemalloc
-)
-
 if (DESKTOP_APP_USE_ALLOCATION_TRACER)
     target_link_options(common_options
     INTERFACE
@@ -114,19 +108,6 @@ if (NOT DESKTOP_APP_USE_PACKAGED)
         -fwhole-program
         -Wl,-z,relro
         -Wl,-z,now
-        # -pie # https://gitlab.gnome.org/GNOME/nautilus/-/issues/1601
-    )
-endif()
-
-if (DESKTOP_APP_USE_PACKAGED)
-    find_library(ATOMIC_LIBRARY atomic)
-else()
-    find_library(ATOMIC_LIBRARY libatomic.a)
-endif()
-
-if (ATOMIC_LIBRARY)
-    target_link_libraries(common_options
-    INTERFACE
-        ${ATOMIC_LIBRARY}
+        -pie
     )
 endif()
